test/unit/year_range_test.rb

```diff
@@ -2,16 +2,50 @@
 
 module SmartAnswer
   class YearRangeTest < ActiveSupport::TestCase
-    setup do
-      @year_range = YearRange.new(begins_on: Date.parse('2000-02-01'))
+    context 'beginning on 7th June, 2001' do
+      setup do
+        @year_range = YearRange.new(begins_on: Date.parse('2001-06-07'))
+      end
+
+      should 'begin on begins_on date' do
+        assert_equal Date.parse('2001-06-07'), @year_range.begins_on
+      end
+
+      should 'end a day before a year after the begins_on date' do
+        assert_equal Date.parse('2002-06-06'), @year_range.ends_on
+      end
+
+      should 'be 365 days long, because it does not include 29th Feb' do
+        assert_equal 365, @year_range.number_of_days
+      end
     end
 
-    should 'begin on begins_on date' do
-      assert_equal Date.parse('2000-02-01'), @year_range.begins_on
+    context 'beginning on 1st February, 2000' do
```

Comment:
I wonder whether it might be a bit clearer if the context was "beginning on 1st February in a leap year"?